homeassistant/components/nederlandse_spoorwegen/sensor.py

```diff
@@ -57,6 +63,146 @@
 )
 
 
+@dataclass(frozen=True, kw_only=True)
+class NSSensorEntityDescription(SensorEntityDescription):
+    """Describes Nederlandse Spoorwegen sensor entity."""
+
+    entity_category: EntityCategory | None = None
+    data_fn: Callable[[Any], Any] | None = None
+    value_fn: Callable[[Any], Any] | None = None
+
+
+# Entity descriptions for all the different sensors we create per route
+SENSOR_DESCRIPTIONS: tuple[NSSensorEntityDescription, ...] = (
+    NSSensorEntityDescription(
+        key="departure",
+        translation_key="departure",
+        device_class=SensorDeviceClass.TIMESTAMP,
+        name="Departure time",
+        icon="mdi:train",
```

Comment:
if we have a translation key, we can remove the name, and move the icon to `icons.json`